cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java

```diff
@@ -243,6 +260,76 @@ private void setRootOutputDirectoryWithPrompt() {
 
   }
 
+  /**
+   * Uses default initialization, checks if monolithic templates exist, handles the exception and lets the user decide
+   * if the templates should be upgraded.
+   */
+  private void checkMonolithicConfigurationException() {
+
+    try {
+      CobiGenUtils.initializeCobiGen(this.templatesProject, false);
+    } catch (DeprecatedMonolithicConfigurationException e) {
+      LOG.warn("Found monolithic configuration at: {} {}",
+          DeprecatedMonolithicConfigurationException.getMonolithicConfiguration(), e.getMessage());
+
+      if (this.upgradeConfiguration || askUserToUpgradeTemplates()) {
+        startTemplatesUpgrader();
+      } else {
+        askUserToPostponeUpgrade();
+      }
+    }
+
+  }
+
+  /**
+   * Opens a looping prompt with a yes/no question and upgrades the templates to the newest version.
+   *
+   */
+  private boolean askUserToUpgradeTemplates() {
+
+    LOG.info(
+        "Would you like to upgrade your templates to the newest version? \n"
+            + MessagesConstants.YES_NO_ANSWER_DESCRIPTION + "For more Informations, please visit: ",
+        WikiConstants.WIKI_UPGRADE_MONOLITHIC_CONFIGURATION, System.getProperty("user.dir"));
+
+    return ValidationUtils.yesNoPrompt("Upgrading templates configuration...: ",
+        MessagesConstants.INVALID_YES_NO_ANSWER_DESCRIPTION, "Continue generation with monolithic templates...");
+  }
+
+  /**
+   * Upgrades the given template configuration.
+   *
+   */
+  private void startTemplatesUpgrader() {
+
+    try {
+      // set the new template-set as the templatesProject after the upgrade
```

Comment:
this can be added to the javadoc and removed

Reply:
done.